pkg/controllers/management/auth/globalroles/globalrolebinding_handler.go

```diff
@@ -137,6 +162,109 @@ func (grb *globalRoleBindingLifecycle) deleteAdminBinding(obj *v3.GlobalRoleBind
 	return nil
 }
 
+// reconcileClusterPermissions grants permissions for the binding in all downstream (non-local) clusters. Will also
+// remove invalid bindings (bindings not for active RoleTemplates or for invalid subjects).
+func (grb *globalRoleBindingLifecycle) reconcileClusterPermissions(globalRoleBinding *v3.GlobalRoleBinding) error {
+	globalRole, err := grb.grLister.Get("", globalRoleBinding.GlobalRoleName)
+	if err != nil {
+		return fmt.Errorf("unable to get globalRole %s: %w", globalRoleBinding.Name, err)
+	}
+	// no per-cluster permissions if the associated globalRole doesn't inherit any clusterRoles
+	if len(globalRole.InheritedClusterRoles) == 0 {
+		return nil
+	}
+	clusters, err := grb.clusterLister.List("", labels.Everything())
```

Comment:
Would we be able to filter the local cluster out via label? Even if we can, I think we should also keep the validation in `L184`.

Reply:
I don't think so. Looking at the local cluster object on my local (basic docker install), the only label that I saw was `provider.cattle.io: k3s`, which appears to be insufficient to identify the local cluster uniquely. Even if I could use some sort of built-in name selector (not sure if there is one), given that we are keeping the if check I think this is of little/no value.

In short, I would prefer to leave this as-is.